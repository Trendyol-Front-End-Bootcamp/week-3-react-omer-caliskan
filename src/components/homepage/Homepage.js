import React, {useEffect, useState} from 'react';
import CharacterList from '../character/CharacterList';
import PageNotFound from '../notfound/PageNotFound';
import '../../assets/css/homepage.css';
import InfiniteScroll from 'react-infinite-scroll-component';
import Loading from '../layout/Loading';
import { getCharactersByPageandFilter } from "../../service/RickandMortyService"

function Homepage({test}){
const [gender, setGender] = useState("");
const [status, setStatus] = useState("");
const [characters, setCharacters] = useState([]);
const [error, setError] = useState(true);
const [page, setPage] = useState(1);
const [hasMore, setHasMore] = useState(true);
const [loading, setLoading] = useState(true);

    React.useEffect(() => {
         const fetch = async () => {
            try {
                test ? setLoading(true) : setLoading(false)
                const data = await getCharactersByPageandFilter(page, gender, status);
                setCharacters([...characters, ...data.results])
                setError(true)
                setHasMore(data.info.next)
                setLoading(true)
            } catch (error) {
                setError(false)
            }
            
        }
         
        fetch();
    }, [gender, status, page]);

    return(
        loading ? <div>
        <div className="FilterBar">
            <div>
                <span>Gender:</span>  
                <select onChange={(s) => {setGender(s.target.value); setPage(1); setCharacters([])}}>
                    <option>All</option>
                    <option>Male</option>
                    <option>Female</option>
                    <option>Genderless</option>
                    <option>Unknown</option>
                    <option>ForTest404</option>
                </select>
            </div>

            <div>
             <span>Status:</span>
            <select onChange={(s) => {setStatus(s.target.value); setPage(1); setCharacters([])}}>
                <option>All</option>
                <option>Alive</option>
                <option>Dead</option>
                <option>Unknown</option>
            </select>
        </div>

        </div>
        { 
            
        error ? <InfiniteScroll
            dataLength={characters.length}
            next={() => setPage(page+1)}
            hasMore={hasMore}
            loader={<Loading />}
            > <CharacterList characters={characters} /> </InfiniteScroll> : <PageNotFound />
        }
        </div> : <Loading />
    );
}

export default Homepage;