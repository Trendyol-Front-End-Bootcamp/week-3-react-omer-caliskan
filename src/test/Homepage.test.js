import React from 'react'
import Homepage from "../components/homepage/Homepage";
import Enzyme, { shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";


Enzyme.configure({ adapter: new Adapter() })

describe("Homepage", () => {
  let homepage
  
  beforeEach(async () => {
    homepage = shallow(<Homepage />)
    })

    it("should render homepage", () => {
      expect(homepage).toMatchSnapshot()
      })

      it("should selects have onchange func", () => {
        const selects = homepage.find('select').at(0).props('onChange')

        expect(selects).toBeTruthy()
      })

    it("should check select items ", () => {
      const select = homepage.find('select').at(0)

      const selectedItems = select.props().children[1].props.children

      /*
      select.simulate('change', {
        target: { value: 'Male' },
      });
      */

      expect(selectedItems).toEqual("Male")

      })
  })