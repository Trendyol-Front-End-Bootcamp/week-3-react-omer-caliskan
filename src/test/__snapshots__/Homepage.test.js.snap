// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage should render homepage 1`] = `ShallowWrapper {}`;

exports[`Homepage should render selects 1`] = `ShallowWrapper {}`;
